staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -70,6 +73,25 @@ func init() {
 // legacyAPIServiceName is the fixed name of the only non-groupified API version
 const legacyAPIServiceName = "v1."
 
+type ServiceResolver interface {
+	ResolveEndpoint(namespace, name string) (*url.URL, error)
```

Comment:
> what would serviceDestinationHost mean?

It's the current value (before this pull) being passed through.  Passing it along means that in the case of using a static mapping, it can be computed once.